fia_api/scripts/transforms/loq_transform.py

```diff
@@ -27,28 +28,48 @@ def apply(self, script: PreScript, job: Job) -> None:  # noqa: C901
+            ):
                 continue
-            if self._replace_input(line, lines, index, "sample_transmission", job.inputs["scatter_transmission"]):  # type: ignore
+            if "scatter_transmission" in job.inputs and self._replace_input(
+                line, lines, index, "sample_transmission", job.inputs["scatter_transmission"]
+            ):
                 continue
-            if self._replace_input(line, lines, index, "sample_direct", job.inputs["scatter_direct"]):  # type: ignore
+            if "scatter_direct" in job.inputs and self._replace_input(
+                line, lines, index, "sample_direct", job.inputs["scatter_direct"]
+            ):
                 continue
-            if self._replace_input(line, lines, index, "can_scatter", job.inputs["can_scatter"]):  # type: ignore
+            if "can_scatter" in job.inputs and self._replace_input(
+                line, lines, index, "can_scatter", job.inputs["can_scatter"]
+            ):
                 continue
-            if self._replace_input(line, lines, index, "can_transmission", job.inputs["can_transmission"]):  # type: ignore
+            if "can_transmission" in job.inputs and self._replace_input(
+                line, lines, index, "can_transmission", job.inputs["can_transmission"]
+            ):
                 continue
-            if self._replace_input(line, lines, index, "can_direct", job.inputs["can_direct"]):  # type: ignore
+            if "can_direct" in job.inputs and self._replace_input(
+                line, lines, index, "can_direct", job.inputs["can_direct"]
+            ):
                 continue
-            if self._replace_input(line, lines, index, "sample_thickness", job.inputs["sample_thickness"]):  # type: ignore
+            if "sample_thickness" in job.inputs and self._replace_input(
+                line, lines, index, "sample_thickness", job.inputs["sample_thickness"]
+            ):
                 continue
-            if self._replace_input(line, lines, index, "sample_geometry", '"' + job.inputs["sample_geometry"] + '"'):  # type: ignore
+            if "sample_geometry" in job.inputs and self._replace_input(
+                line, lines, index, "sample_geometry", '"' + job.inputs["sample_geometry"] + '"'
+            ):
                 continue
-            if self._replace_input(line, lines, index, "sample_height", job.inputs["sample_height"]):  # type: ignore
+            if "sample_height" in job.inputs and self._replace_input(
+                line, lines, index, "sample_height", job.inputs["sample_height"]
+            ):
                 continue
-            if self._replace_input(line, lines, index, "sample_width", job.inputs["sample_width"]):  # type: ignore
+            if "sample_width" in job.inputs and self._replace_input(
+                line, lines, index, "sample_width", job.inputs["sample_width"]
+            ):
                 continue
```

Comment:
Could tidy this up and use the same logic you've used in the test file with a dictionary, a loop, and enumerate.